utils/result.hh

```diff
@@ -44,4 +44,15 @@ concept ExceptionContainerResult = bo::is_basic_result<R>::value && ExceptionCon
 template<typename F>
 concept ExceptionContainerResultFuture = seastar::is_future<F>::value && ExceptionContainerResult<typename F::value_type>;
 
+template<typename L, typename R>
+concept ResultCompatible =
+    bo::is_basic_result<L>::value &&
+    bo::is_basic_result<R>::value &&
+    std::same_as<typename L::error_type, typename R::error_type> &&
+    std::same_as<typename L::no_value_policy_type, typename R::no_value_policy_type>;
+
+// Creates a compatible result type which can hold value type T or the same exceptional container as R
+template<typename T, ExceptionContainerResult R>
+using compatible_result = bo::result<T, typename R::error_type, exception_container_throw_policy>;
```

Comment:
In std::allocator terminology, this is a rebind - we rebind the allocator from one type to another, but retaining the allocation method.

So I suggest to call this rebind_result. The value of reusing terminology is low when the terminology reused is obscure, but non-zero.

Let's also make the ResultCompatible concept less vague. I don't have a good suggestion though.

Reply:
Done, changed `compatible_result` -> `rebind_result` and `ResultCompatible` -> `ResultRebindableTo`.